src/main/java/seedu/address/model/loan/DataField.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.model.loan;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * Represents a generic field that is directly describable as a string (called an objString).
+ * T: The type of the underlying value this DataField has.
```

Comment:
Redundant

Reply:
`T` is just a type, but of what? What significance does `T` have: is it the type of a member, a parameter of a method etc.? The name "T" itself doesn't say anything.